service/dps/storm/topologies/indexing/src/main/java/eu/europeana/cloud/service/dps/storm/topologies/indexing/IndexingTopology.java

```diff
@@ -0,0 +1,174 @@
+import org.apache.storm.spout.SchemeAsMultiScheme;
+import org.apache.storm.topology.TopologyBuilder;
+import org.apache.storm.tuple.Fields;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * Created by pwozniak on 4/6/18
+ */
+public class IndexingTopology {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(IndexingTopology.class);
+
+    private static Properties topologyProperties = new Properties();
+    private static Properties indexingProperties = new Properties();
+    private final BrokerHosts brokerHosts;
+    private static final String TOPOLOGY_PROPERTIES_FILE = "indexing-topology-config.properties";
+    private static final String INDEXING_PROPERTIES_FILE = "indexing.properties";
+    private final String DATASET_STREAM = InputDataType.DATASET_URLS.name();
+    private final String FILE_STREAM = InputDataType.FILE_URLS.name();
+    public static final String SUCCESS_MESSAGE = "Record is indexed correctly";
+
+    public IndexingTopology(String defaultPropertyFile, String providedPropertyFile, String defaultIndexingPropertiesFile, String providedIndexingPropertiesFile) {
+        PropertyFileLoader.loadPropertyFile(defaultPropertyFile, providedPropertyFile, topologyProperties);
+        PropertyFileLoader.loadPropertyFile(defaultIndexingPropertiesFile, providedIndexingPropertiesFile, indexingProperties);
+        brokerHosts = new ZkHosts(topologyProperties.getProperty(TopologyPropertyKeys.INPUT_ZOOKEEPER_ADDRESS));
+    }
+
+    public final StormTopology buildTopology(String indexingTopic, String ecloudMcsAddress) {
+        Map<String, String> routingRules = new HashMap<>();
+        routingRules.put(DATASET_STREAM, DATASET_STREAM);
+        routingRules.put(FILE_STREAM, FILE_STREAM);
+        ReadFileBolt retrieveFileBolt = new ReadFileBolt(ecloudMcsAddress);
+
+        SpoutConfig kafkaConfig = new SpoutConfig(brokerHosts, indexingTopic, "", "storm");
+        kafkaConfig.scheme = new SchemeAsMultiScheme(new StringScheme());
+        kafkaConfig.ignoreZkOffsets = true;
+        kafkaConfig.startOffsetTime = kafka.api.OffsetRequest.LatestTime();
+        TopologyBuilder builder = new TopologyBuilder();
+        KafkaSpout kafkaSpout = new CustomKafkaSpout(kafkaConfig, topologyProperties.getProperty(TopologyPropertyKeys.CASSANDRA_HOSTS),
+                Integer.parseInt(topologyProperties.getProperty(TopologyPropertyKeys.CASSANDRA_PORT)),
+                topologyProperties.getProperty(TopologyPropertyKeys.CASSANDRA_KEYSPACE_NAME),
+                topologyProperties.getProperty(TopologyPropertyKeys.CASSANDRA_USERNAME),
+                topologyProperties.getProperty(TopologyPropertyKeys.CASSANDRA_SECRET_TOKEN));
+
+
+        builder.setSpout(TopologyHelper.SPOUT, kafkaSpout,
+                ((int) Integer.parseInt(topologyProperties.getProperty(TopologyPropertyKeys.KAFKA_SPOUT_PARALLEL))))
```

Comment:
1- There is no need for the casting (int)
2- This line with Integer.parseInt(topologyProperties.getProperty(TopologyPropertyKeys.CASSANDRA_PORT): will generate a major sonar bug : "Method converts String to boxed primitive using excessive boxing"
I would suggest to extract this line in a method and calling the method. In this case you will create one bug instead of 17. Note: The entire bug should be resolved but maybe in a different task.